### PR TITLE
feat: add openfga to compose

### PR DESCRIPTION
**Pull request by ryanfang5**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -178,6 +178,24 @@ services:
     profiles:
       - all
       - backend
+  openfga:
+    environment:
+      OPENFGA_LISTOBJECTSMAXRESULTS: 10000
+      OPENFGA_LISTOBJECTSDEADLINE: "10s"
+      OPENFGA_LOG_FORMAT: "json"
+    image: openfga/openfga:latest
```
Comment (igboyes) [inline docker-compose.yml:186]: Please pin to the latest version number. We don't want any surprises when `latest` starts meaning something else.

File: docker-compose.yml [hunk 2]
```diff
@@ -178,6 +178,24 @@ services:
...
+    container_name: openfga
+    command: run --datastore-engine postgres --datastore-uri 'postgres://virtool:virtool@postgres:5432/postgres?sslmode=disable'
```
Comment (igboyes) [inline docker-compose.yml:188]: Is this sticking the data in the `virtool` database along with our application data? I think the OFGA and VT data should be in separate DBs in the same Postgres server.

**Review by igboyes — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (ReeceHoffmann) [inline docker-compose.yml:188]: Not strictly required in dev, but I would move the `--datastore-engine` and `--datastore-uri` commands to be passed as env variables. Presumably in prod the secure DB password is going to be provided via env variable or similar.

https://openfga.dev/docs/getting-started/setup-openfga

File: docker-compose.yml [hunk 3]
```diff
@@ -178,6 +178,24 @@ services:
...
+    networks:
+      - default
+    ports:
+      - "8080:8080"
+      - "3000:3000"
```
Comment (ReeceHoffmann) [inline docker-compose.yml:193]: I think it makes sense to expose the 3000 playground port in dev, so we should leave this as is, but just as a note, the docs recommend against exposing 3000 in prod

File: docker-compose.yml [hunk 4]
```diff
@@ -178,6 +178,24 @@ services:
...
+    profiles:
+      - all
+      - test
+      - backend
+      - ui
```
Comment (ReeceHoffmann) [inline docker-compose.yml:198]: We should also add a depends_on field to ensure this doesn't try to start prior to PostgreSQL starting. Something like what we have for the server: 

```
depends_on:
      - postgres
```

File: docker-compose.yml [hunk 5]
```diff
@@ -178,6 +178,24 @@ services:
     profiles:
       - all
       - backend
+  openfga:
+    environment:
+      OPENFGA_LISTOBJECTSMAXRESULTS: 10000
+      OPENFGA_LISTOBJECTSDEADLINE: "10s"
+      OPENFGA_LOG_FORMAT: "json"
+    image: openfga/openfga:latest
+    container_name: openfga
```
Comment (ReeceHoffmann) [inline docker-compose.yml:187]: Do we need to specify `container_name`? Wouldn't it generate something similar using the name provided on line 181?

File: docker-compose.yml [hunk 6]
```diff
@@ -178,6 +178,24 @@ services:
...
+    command: run --datastore-engine postgres --datastore-uri 'postgres://virtool:virtool@postgres:5432/postgres?sslmode=disable'
+    networks:
+      - default
```
Comment (ReeceHoffmann) [inline docker-compose.yml:190]: Do we need to specify the default networks explicitly?

**Review by ReeceHoffmann — CHANGES_REQUESTED** [review]
Mostly looks good! I also think we should probably also add the migrate service to the compose file. Looks to me like the migrate command has to be run at least once on a given postgres server before OFGA can interact with it. Might also be important for OFGA updates, but docs are not clear on that.

https://openfga.dev/docs/getting-started/setup-openfga

**Comment by ryanfang5** [comment]
Took way too long due to not attaching persisting volume to the postgres-openfga container. Also moved commands into environment variables and removed default networks. Let me know if any naming should be changed.

**Review by igboyes — APPROVED** [review]

**Pull request merged**